src/operation-node/json-path-reference-node.ts

```diff
@@ -0,0 +1,46 @@
+import { freeze } from '../util/object-utils.js'
+import { JSONPathNode } from './json-path-node.js'
+import { OperationNode } from './operation-node.js'
+import { JSONOperator } from './operator-node.js'
+
+type JSONOperatorNo$ = Exclude<JSONOperator, '->>$'>
```

Comment:
Should also exclude `->$` right?

Reply:
Nice catch!

Fixed @ 713e97e5e955a333d12970f508d4cf85dfc54925.